internal/gcsx/read_manager/read_manager.go

```diff
@@ -35,19 +35,51 @@ type ReadManager struct {
 	readers []gcsx.Reader
 }
 
-// NewReadManager creates a new ReadManager for the given GCS object.
-// It initializes the manager with a file cache reader and a GCS reader.
-func NewReadManager(o *gcs.MinObject, bucket gcs.Bucket, sequentialReadSizeMB int32, fileCacheHandler *file.CacheHandler, cacheFileForRangeRead bool, metricHandle common.MetricHandle, mrdWrapper *gcsx.MultiRangeDownloaderWrapper) *ReadManager {
-	gcsReader := clientReaders.NewGCSReader(o, bucket, metricHandle, mrdWrapper, sequentialReadSizeMB)
-	fileCacheReader := gcsx.NewFileCacheReader(o, bucket, fileCacheHandler, cacheFileForRangeRead, metricHandle)
+// ReadManagerConfig holds the configuration parameters for creating a new ReadManager.
+type ReadManagerConfig struct {
```

Comment:
nit: You may separate common variable and reader specific variable for readability.